src/parser.lisp

```diff
@@ -1,3 +1,63 @@
 ;;;; Parser of scene files (.obj).
 
 (in-package #:cl-raster/parser)
+
+(defun raw-line-to-tokens (line)
+  (let ((line-no-comments (str:trim
+                            (str:substring
+                              0
+                              (position #\# line)
+                              line))))
+    (str:split " " line-no-comments :omit-nulls T)))
+
+(defun parse-mtl-file (filename)
+  (let ((materials (make-hash-table))
+        (current-mtl NIL))
+    (with-open-file (in-stream filename)
+      (loop for line = (read-line in-stream NIL)
+            while line
+            for tokens = (raw-line-to-tokens line)
+            when tokens
+            do (progn
+                 (when (eq (first tokens) "newmtl")
+                   (setf (gethash (second tokens) (make-mtl))))
+                 (when (eq (first tokens) "Ks")
+                   (setf (material-Ks (gethash current-mtl))
+                         (list (parse-float (second tokens))
+                               (parse-float (third tokens))
+                               (parse-float (fourth tokens))))))))
+    materials))
+
+;;; See https://en.wikipedia.org/wiki/Wavefront_.obj_file
+
+(defun parsei-obj-file (filename)
+  (let ((scene (list ()))
+        (triangles (list))
+        (points (list))
+        (points-norm (list))
+        (points-texture (list))
+        (materials (make-hash-table)))
+    (with-open-file (in-stream filename)
+      (loop for line = (read-line in-stream NIL)
+            while line
+            for tokens = (raw-line-to-tokens line)
+            when tokens
+            do (progn
+                 (when (eq (first tokens) "mtllib")
+                   (parse-mtl-file (second tokens)))
+                 (when (eq (first tokens) "usemtl")
+                   (setf current-material (gethash (second tokens) materials)))
```

Comment:
What does `current-material` do? It is not used anywhere else and it is not defined anywhere.
If it is going to be a global variable it should have earmuffs: `*current-material*`